### ISSUE TITLE
Deploy to smaller vm sizes (Basic_A1 / Standard_D1 / Standard_D1_V2)

### ISSUE THREAD

**Issue by tekmaven** [Azure/acs-engine#352]
When trying to create a deployment with the following configuration:

```json
"orchestratorProfile": {
  "orchestratorType": "Kubernetes"
},
"agentPoolProfiles": [
  {
    "name": "agentpool1",
    "count": 2,
    "vmSize": "Basic_A1",
    "availabilityProfile": "AvailabilitySet"
  }
],
...
```

We are receiving an error on ARM template validation:

> New-AzureRmResourceGroupDeployment : 4:56:42 PM - Error: Code=InvalidTemplate; Message=Deployment template validation failed: 'The provided value 'Basic_A1' for the template parameter 'agentpool1VMSize' at line '123' and column '30' is not valid. The parameter value is not part of the allowed value(s): 'Basic_A2,Basic_A3,Basic_A4,Standard_A10,Standard_A11,Standard_A2,Standard_A2_v2,Standard_A2m_v2,Standard_A3,Standard_A4,Standard_A4_v2,Standard_A4m_v2,Standard_A5,Standard_A6,Standard_A7,Standard_A8,Standard_A8_v2,Standard_A8m_v2,Standard_A9,Standard_D11,Standard_D11_v2,Standard_D12,Standard_D12_v2,Standard_D13,Standard_D13_v2,Standard_D14,Standard_D14_v2,Standard_D15_v2,Standard_D2,Standard_D2_v2,Standard_D3,Standard_D3_v2,Standard_D4,Standard_D4_v2,Standard_D5_v2,Standard_DS11,Standard_DS11_v2,Standard_DS12,Standard_DS12_v2,Standard_DS13,Standard_DS13_v2,Standard_DS14,Standard_DS14_v2,Standard_DS15_v2,Standard_DS2,Standard_DS2_v2,Standard_DS3,Standard_DS3_v2,Standard_DS4,Standard_DS4_v2,Standard_DS5_v2,Standard_F16,Standard_F16s,Standard_F2,Standard_F2s,Standard_F4,Standard_F4s,Standard_F8,Standard_F8s,Standard_G1,Standard_G2,Standard_G3,Standard_G4,Standard_G5,Standard_GS1,Standard_GS2,Standard_GS3,Standard_GS4,Standard_GS5,Standard_H16,Standard_H16m,Standard_H16mr,Standard_H16r,Standard_H8,Standard_H8m,Standard_NC12,Standard_NC24,Standard_NC24r,Standard_NC6,Standard_NV12,Standard_NV24,Standard_NV6'.'.
```

For a cluster with small amounts of pods which require low resources, A1 or D1 nodes make a lot of sense.  Also, we are exploring having a large amount of small nodes vs few large nodes.

**Comment by colemickens** [comment]
We intentionally block creation of clusters with nodes that small: https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/Get-AzureConstants.ps1#L40

You could modify that PowerShell script, enable the smaller size, run the script, rebuild ACS-Engine and get the desired effect. I don't think we want to support this in ACS-Engine however. Those VMs are simply too under-powered for most real world cases.

I'll leave this open so @anhowe can take a look when he's back. Also, we really need to convert that into something other than a PowerShell script...

**Comment by tekmaven** [comment]
I figured that was the intention.

Deploying a 6 node A1 cluster vs a 3 node A2 cluster will yield the same amount of RAM for the cluster at 1/3rd less cost.

**Comment by tekmaven** [comment]
A great workaround from a teammate - 

Deploy a Basic_A2 and then resize to A1.  Testing this now.

**Comment by colemickens** [comment]
While it might be the same amount of RAM, nearly everything else is different - different amounts of allowed attached disk, different performance characteristics, and very different real world behavior given that there is a fixed per-node overhead that is more pronounced on small SKUs. To put it another way, this limitation was added as a direct result of customer issues on smaller SKUs.

The workaround is a great idea; wish I'd thought of it. You should be able to use that technique reliably for both the master nodes and agent nodes. Let us know how it goes.

**Comment by yangl900** [comment]
I have been running both Basic_A0 and Standard_A1 for a while. They have slower in network and less CPU power, but works fine for testing environment. I'd suggest we give warning saying this should not use in production, but should allow user to use smaller size VMs.

**Comment by anhowe** [comment]
@shrutir25 fixed this in #454.  We took out Basic, due to its lack of support for LB, and other Azure features.  There are equivalent and cheaper sizes than basic.  Please checkout the latest "promo" sizes.
